### DISCUSSION TITLE
Looking for a simple (automated) Hg-like workflow

### DISCUSSION THREAD

**Discussion by zoharl3** [community/community#22950] [New to GitHub]
<p>I asked here a basic question, but it wasn’t answered:</p>
<p><a href="https://stackoverflow.com/questions/54850153/git-looking-for-a-simple-hg-like-workflow" rel="nofollow noopener">https://stackoverflow.com/questions/54850153/git-looking-for-a-simple-hg-like-workflow</a></p>

**Comment by AndreaGriffiths11** [comment]
<p>Hey <a class="mention" href="/zoharl3">@zoharl3</a>,</p>
<p>Thanks for being here! The link to stack is null now, would you mind sharing what you are trying to accomplish, perhaps then we can help point you on the right direction.</p>

**Comment by zoharl3** [comment]
<p>Copying the link content:</p>
<p>There’s a repository that is updated from 3 places:</p>
<ul>
<li>
<p>a home computer</p>
</li>
<li>
<p>a work computer</p>
</li>
<li>
<p>a friend</p>
</li>
</ul>
<p>I’m used to save (backup) my code changes often (every hour), and I also use the repository to sync between my computers (I’m just running a script).</p>
<p>In Hg, it’s simple, and there were no problems so far:</p>
<pre><code>hg commit -R %s -m "." -v

hg fetch -R %s -v

hg push -R %s
</code></pre>
<p>where %s stands for the harddrive directory.</p>
<p>In git, I’m doing:</p>
<pre><code>git -C %s commit -a -m "." -v

git -C %s pull -v

git -C %s push
</code></pre>
<p>However, it seems to have issues:</p>
<ul>
<li>
<p>Files that I didn’t touch are merged and reported as me changing them.</p>
</li>
<li>
<p>Sometimes, line endings are changed, and white spaces are added.</p>
</li>
<li>
<p>Sometimes, something fails, e.g. commit isn’t done, not sure what, I’m not keeping a log, but it seems like this workflow is unreliable.</p>
</li>
</ul>
<p>What’s the suggested foolproof, hands off workflow for git?</p>

**Comment by lee-dohm** [comment, marked as answer]
<p>If line endings are changing, then it sounds like something isn’t configured properly. And if that assumption is correct, then it may be causing some or all of the other problems. You may want to consult our <a href="https://help.github.com/en/articles/dealing-with-line-endings" rel="nofollow noopener">help documentation on dealing with line endings in git</a>.</p>
<p>If that doesn’t fix it, then if you can give us some steps to reproduce the problem, we can take a deeper look. Please include:</p>
<ul>
<li>A starting state (preferably a commit in a public repository)</li>
<li>The exact steps you took</li>
<li>What you expected to happen</li>
<li>What actually happened</li>
</ul>
<p>Thanks!</p>

**Comment by zoharl3** [comment]
<p>This sounds like a promising direction to look into and a fair request (to reproduce the problem). Since it may take me some time to look into it, I’m accepting this as the solution for now.</p>
<p>You also confirmed that my pipeline should work fine in general.</p>
<p>Thanks</p>

**Comment by zoharl3** [comment]
<p>I think I found the issue: my TortoiseGit was configured to use cygwin git instead of PortableGit. The former has issues with line endings and reports everything as modified. GUI clients actually warn against using cygwin git.</p>
